src/plots/cartesian/layout_attributes.js

```diff
@@ -279,6 +279,59 @@ module.exports = {
         role: 'style',
         description: 'Determines whether or not the tick labels are drawn.'
     },
+    showspikes: {
+        valType: 'boolean',
+        dflt: false,
+        role: 'style',
+        description: [
+            'Determines whether or not spikes (aka droplines) are drawn for this axis.',
+            'Note: This only takes affect when hovermode = closest'
+        ].join(' ')
+    },
+    spikecolor: {
+        valType: 'color',
+        dflt: null,
+        role: 'style',
+        description: 'Sets the spike color. If undefined, will use the series color'
+    },
+    spikethickness: {
+        valType: 'number',
+        dflt: 3,
+        role: 'style',
+        description: 'Sets the width (in px) of the zero line.'
+    },
+    spikedash: {
+        valType: 'string',
+        // string type usually doesn't take values... this one should really be
+        // a special type or at least a special coercion function, from the GUI
+        // you only get these values but elsewhere the user can supply a list of
+        // dash lengths in px, and it will be honored
+        values: ['solid', 'dot', 'dash', 'longdash', 'dashdot', 'longdashdot'],
```

Comment:
Non- :no_entry: , but It would be nice to reuse the list in `traces/scatter/attributes.js`